holoviews/core/data/image.py

```diff
@@ -0,0 +1,250 @@
+import numpy as np
+
+from ..boundingregion import BoundingRegion, BoundingBox
+from ..dimension import Dimension
+from ..element import Element
+from ..ndmapping import OrderedDict, NdMapping, item_check
+from ..sheetcoords import SheetCoordinateSystem, Slice
+from .. import util
+from .grid import GridInterface
+from .interface import Interface
+
+
+class ImageInterface(GridInterface):
+    """
+    Interface for 2 or 3D arrays representing images
+    of raw luminance values, RGB values or HSV values.
+    """
+
+    types = (np.ndarray,)
+
+    datatype = 'image'
+
+    @classmethod
+    def init(cls, eltype, data, kdims, vdims):
+        if kdims is None:
+            kdims = eltype.kdims
+        if vdims is None:
+            vdims = eltype.vdims
+
+        kwargs = {}
+        dimensions = [d.name if isinstance(d, Dimension) else
+                      d for d in kdims + vdims]
+        if isinstance(data, tuple):
+            data = dict(zip(dimensions, data))
+        if isinstance(data, dict):
+            xs, ys = np.asarray(data[kdims[0].name]), np.asarray(data[kdims[1].name])
+            l, r, xdensity, invertx = util.bound_range(xs, None)
+            b, t, ydensity, inverty = util.bound_range(ys, None)
```

Comment:
Will need replacing (see comment below). I think that at least l,r,b and t can be computed without resorting to ``np.diff``. The densities too I assume.